### DISCUSSION TITLE
Website field in favorites and import GPX links

### DISCUSSION THREAD

**Discussion by GpxFeed** [devemux86/cruiser#565] [Q&A] [answered]
We've already talked about it twice - maybe that was in the last millennium ??? ... I forgot ...

The [GPS Exchange Format](https://de.wikipedia.org/wiki/GPS_Exchange_Format) supports a link tag.

	<link>URL</link>

Copy&Paste wouldn't be bad in this context either.

Up to now only *GPXSee* or a text editor are getting me out of the jam.

It would be great if you could at least support this feature(s) for tour planning issues in your Java version.

-- 
Brinki

**Comment by devemux86** [comment]
@GpxFeed thanks for the suggestion.

We currently display the name, description and other data, such as elevation, for information purposes.

**Reply by GpxFeed** [reply]
The first screenshot, taken with 'GPXSee', shows what I asked for quite some time now.

The second screenshot, taken with 'Cruiser', shows what it still is.

When planning a tour with the Desktop Cruiser, if there is a link tag, I would like to be able to open this website in Firefox or a similar browser with a single click on the link below the text.

Copy and paste wouldn't be bad either.

GPX-Data-Source:

https://github.com/GpxFeed/campgrounds/raw/refs/heads/master/gpx/france-campsites.gpx

<img width="1919" height="1051" alt="Screenshot-GPXSee" src="https://github.com/user-attachments/assets/7d8f8849-ecb0-43fd-b1a5-2c246cf9ed00" />

<img width="1920" height="1050" alt="Screenshot-Cruiser" src="https://github.com/user-attachments/assets/46eb5a07-5d91-4b03-9a8f-e702bd9e54d6" />

**Comment by devemux86** [comment, marked as answer]
- Implemented in https://github.com/devemux86/cruiser/discussions/891

**Comment by 0709wiwiwi** [comment]
@GpxFeed 

It seems you may also expect that 'links" in the gpx desc element are  clickable?
Additonally in the wpt desc, as the default gpx wpt link is already offered in the **mobile** app. 
Probably it is too well "hidden" even in the (mobile) Cruiser app to be reallly comfortable ?

Tap and go to the website. Right? 
Locus and GPX Viewer offer this, as well bu the gpx standard link and even for those links that are attached in the wpt desc.

Note: 
The original gpx file containing the multiple gpx wpt's is invalid.
Someone should so please notify the GpxFeed developers.

The solution is simple. See the images.  
The Notepad++ xml schema validator did effictively and exactly show the missing attribute author issue.
But this  same validator did not indicate the trouble hapening with the license element.  

You need to gind out what is causing this particular issue here: 
http://www.topografix.com/GPX/1/1/#type_copyrightType
Tap the question mark after the license element and see:

**Element: license**
_Link to external file containing license text._


Non valid.

<img width="787" height="334" alt="Non valid gpx" src="https://github.com/user-attachments/assets/4779d7ea-7ef0-4ada-81f4-ccf16ae3d701" />

-
Valid.

<img width="818" height="197" alt="image" src="https://github.com/user-attachments/assets/518df1fd-4591-4ea6-ba10-94f847d08a77" />
-

And now at last ...pfff.....even these so EXTREMELY VERY STRICT Garmin programs like old Mapsource does not reject the next import.  ;-)

[Uploading france-campsites_all.gpx.txt…]()

**Comment by GpxFeed** [comment]
Moin,

I think we’re talking at cross-purposes.

In this context, I'm not interested at all in the GPX description tag (Desc) – I'm only interested in a working link tag in the Cruiser desktop version and not so much in the Android version.

Unlike ‘GPXSee’, none of the examples in the attached ZIP file for ‘Cruiser’ provides access to the desired link.

**Reply by 0709wiwiwi** [reply]
“I'm only interested in a working link tag in the Cruiser desktop version, not so much in the Android version.”

You might not be, but others probably are. It would be useful if this could be used in both the desktop and Android versions in the most elegant way possible, just like in your GPXSee example. 

In Android, you can already use the GPX (website) link, but only after clicking the “website” function in favorites. 

@devemux86 
Importing the **multiple** camping poi's by the "simple router import menu" causes the Android Cruiser app to hang.

@GpxFeed 
By the way, apparently you are the author of these feeds.  Could you take a look at the validty comment?

**Reply by devemux86** [reply]
Apps can not display thousands of waypoints and tracks.
GPX files are better suited to contain data for specific, smaller areas.

**Comment by devemux86** [comment]
When you import GPX waypoints and they contain a link, you can use their menu to open their website.
(long press on Android and right mouse click on desktop - see the images)

The website field was added in favorites on Android and desktop.
We also need to add it on overlays on desktop in next update.
(when we use the menu "Routing | Import | Overlay")

<img src="https://github.com/user-attachments/assets/08a7d2f3-dc4d-462b-9e4f-70ea36bbc1cf" width="50%"/>
<img src="https://github.com/user-attachments/assets/d4ccaae0-e167-4609-9983-0055f5884bc5"/>

**Reply by 0709wiwiwi** [reply]
Thanks for that extra info Emux.  I will have a look at it.   Desktop: check for the website by a right click ?

Okido found.

_Once you get the hang of it, it's easy
Desktop Cruiser.
Select the Favorites tab > Select Import.
Select and import the GPX file containing all the favorites.
The favorites icons are now displayed onto the map.
Right-click a favorite icon > tap Website. Et voilà, you're all set._

And sorry I did not notice that these multiple tracks at the same gpx import where the trouble makers here. 

@GpxFeed 
I wonder are those tracks necesarry  ? 
What is their purpose ? 
I've also noticed GPX wpt with “way”?
 It all seems a bit messy to me. Can you check this? 

Contains only the gpx wpt:
[france-campsites.gpx.txt](https://github.com/user-attachments/files/28418456/france-campsites.gpx.txt)

Opening france-campsites.gpx 9454 waypoints.  Garmin Mapsource friendly gpx file

**Comment by GpxFeed** [comment]
> I wonder are those tracks necesarry ?
>What is their purpose ?

Overpass-Turbo, also known as OSM, provides many POIs that consist only of a track (TRK) as their boundary but do not have any individual waypoints (WPT).

In this case, Cruiser provides only the boundary line – no further information.

A replacement POI with a waypoint can be generated in this case using the centre of gravity of the enclosed area and a copy of the remaining TRK data.

You can then delete the original POI with the track... or not.

These enclosing tracks are often helpful if you want to assess the location and size of campsites more precisely for your own planning.

Attached is an Overpass Turbo example with TRK only.

[france.gpx.gz](https://github.com/user-attachments/files/28420508/france.gpx.gz)

**Reply by 0709wiwiwi** [reply]
Okay, Roger.  I seem to remember seeing something like that somewhere before—maybe on the Locus forum? Could that have been you?

**Comment by GpxFeed** [comment]
I'm not familiar with the Locus Forum... let's take a look!

One source of information for everything is the GPX standard, e.g.,
‘https://de.wikipedia.org/wiki/GPS_Exchange_Format’.

Countless Python scripts provide ideas and help with processing GPX files.

**Reply by GpxFeed** [reply]
Correction - should read:
https://de.wikipedia.org/wiki/GPS_Exchange_Format

**Reply by 0709wiwiwi** [reply]
Cruiser GPS does indeed pay very close attention to the creation of valid GPX files.

https://github.com/devemux86/cruiser/discussions/250#discussioncomment-16354589

Many GPS programs are, however, quite flexible when it comes to invalid or incorrect GPX files.
Garmin is not at all that friendly. Garmin rejects imports of invalid GPX files without any exception.
Your GPXFeed export file is indeed incorrect.  But ultimately, it’s easy to fix.
I’ve now updated my post above; The result is now even compliant with Garmin programs.
Anyway if the file does indeed import in the strict Garmin programs you're fine.
Since the Notepad++ XML validator didn’t flag the license element issue, it took a bit of searching.

http://www.topografix.com/GPX/1/1/#SchemaProperties

Notice the https in the second half of the schema location.

xsi:schemaLocation="http://www.topografix.com/GPX/1/1 **https**://www.topografix.com/GPX/1/1/gpx.xsd"

_Advised by the gpx standard developper.  Usefully used for faster online validation.  (No extra redirect from http to https)_

**Reply by GpxFeed** [reply]
You can perform a basic validation of GPX files using:

`# tidy --quote-ampersand no -i -xml -utf8 < my-file.gpx`

Now I’ve been playing around a bit with the ‘xmllint’ offline validation using the official online XSD:

Download the latest schema from the web

`# wget -c https://www.topografix.com/GPX/1/1/gpx.xsd`

Validate the GPX file

`# xmllint --schema gpx.xsd my-file.gpx --noout`

I’ll have a look at this when I get the chance one day.

**Reply by 0709wiwiwi** [reply]
Checking a GPX file for validity—especially for those extremely strict Garmin programs—is always a bit of a challenge.
Manual or automatic? Apparently, it can be done with some Windows programs. Is it apparently harder with Linux? Who else can do it? "AI".  Yes, I just tested it. "AI" does indeed find these errors.  And it even recognizes Garmin’s quirks. Generating an "AI" report, however, might use up my free data allowance for today. As soon as the "AI" report can be generated later **and I’m satisfied with it**, I’d like to share it here as well.

Done: https://github.com/devemux86/cruiser/discussions/911#discussioncomment-17140801

**Comment by devemux86** [comment]
There are thousands of camping sites (as points + polygons) in the gpx.

For fast performance do not store a separate `trk` for each camping site.

Instead store a `trkseg` for each camping site and all of them into **1** `trk`.
Then performance is very fast in Cruiser.

Example with 1 `trk` and thousands of `trkseg`: [france-campsites.zip](https://github.com/user-attachments/files/28876516/france-campsites.zip)

**Comment by GpxFeed** [comment]
I've never ever had any issues with the sheer volume of data on Cruiser 
Desktop (Linux).

With the Cruiser app, I cut out the required part I actually need using 
a suitable polygon, just like cutting out cookies for Christmas baking.

I generate the polygon along my planned route and the resulting subset 
of data from the POIs - for example, from the “france-campsites.gpx” 
file - using a Python script.

My script operates across borders as needed.

**Reply by devemux86** [reply]
Desktop and mobile devices are completely different platforms.

**Comment by GpxFeed** [comment]
Am 12.06.26 um 15:34 schrieb Emux:
> Desktops and mobile devices are completely different platforms.

As I mentioned earlier, the feature in the desktop version would be more 
than enough for my planning needs.
